x86 Windows fails to build

In one of the projects we use Parcel, but for some reason, Appveyor x86 fails to build the native modules with this error:
```
error LNK2001: unresolved external symbol "struct std::atomic<bool> Napi::details::needs_objectwrap_destructor_fix"
```

More details:

https://github.com/atom-ide-community/fuzzaldrin-plus-fast/issues/15
https://ci.appveyor.com/project/aminya/fuzzaldrin-plus-fast-se2je


```ps1
> node-gyp-build
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  nothing.c
  win_delay_load_hook.cc
  nothing.vcxproj -> C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map\build\Release\\nothing.lib
  SourceMap.cpp
  MappingContainer.cpp
  MappingLine.cpp
  Mapping.cpp
  win_delay_load_hook.cc
     Creating library C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map\build\Release\sourcemap.lib and object C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map\build\Release\sourcemap.exp
SourceMap.obj : error LNK2001: unresolved external symbol "struct std::atomic<bool> Napi::details::needs_objectwrap_destructor_fix" (?needs_objectwrap_destructor_fix@details@Napi@@3U?$atomic@_N@std@@A) [C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map\build\sourcemap.vcxproj]
C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map\build\Release\sourcemap.node : fatal error LNK1120: 1 unresolved externals [C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map\build\sourcemap.vcxproj]
gyp ERR! build error 
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\projects\\fuzzaldrin-plus-fast-se2je\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\projects\fuzzaldrin-plus-fast-se2je\node_modules\@parcel\source-map
gyp ERR! node -v v14.6.0
gyp ERR! node-gyp -v v6.1.0
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @parcel/source-map@2.0.0-alpha.4.13 install: `node-gyp-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @parcel/source-map@2.0.0-alpha.4.13 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\appveyor\AppData\Roaming\npm-cache\_logs\2020-09-02T05_31_48_721Z-debug.log
Command exited with code 1
```

--- a/azure-pipelines-template.yml
+++ b/azure-pipelines-template.yml
@@ -65,6 +65,10 @@ jobs:
       - script: npm run prebuild
         displayName: Test prebuild
 
+      - script: npm run prebuild-ia32
+        displayName: Test prebuild-ia32
+        condition: eq('${{ parameters.is32bit }}', 'true')
+
       - task: PublishBuildArtifacts@1
         displayName: Publish Native Artifacts
         condition: and(succeeded(), eq(variables['publish_artifacts'], 'true'))

--- a/azure-pipelines-template.yml
+++ b/azure-pipelines-template.yml
@@ -6,6 +6,9 @@ parameters:
 - name: 'container'
   type: string
   default: ''
+- name: 'is32bit'
+  type: string
+  default: 'false'
 
 jobs:
   - job: ${{ parameters.name }}
@@ -25,6 +28,7 @@ jobs:
       - task: NodeTool@0
         inputs:
           versionSpec: $(node_version)
+          force32bit: ${{ parameters.is32bit }}
         displayName: "Install Node.js"
       - script: npm install --build-from-source
         displayName: Install dependencies

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -29,6 +29,12 @@ jobs:
       name: Windows
       vmImage: vs2017-win2016
 
+  - template: azure-pipelines-template.yml
+    parameters:
+      name: Windowsx86
+      vmImage: vs2017-win2016
+      is32bit: true
+
   - ${{ if startsWith(variables['Build.SourceBranch'], 'refs/tags/v') }}:
       - job: Publish
         condition: succeeded()

--- a/bench/run.js
+++ b/bench/run.js
@@ -111,19 +111,21 @@ init.then(() => {
     map.delete();
   });
 
-  suite.add('combine 1000 maps using flatbuffers and stringify', async () => {
-    let map = new SourceMap();
-    for (let i = 0; i < 1000; i++) {
-      map.addBufferMappings(sourcemapBuffer, i * 4);
-    }
-    await map.stringify({
-      file: 'index.js.map',
-      sourceRoot: '/',
-      // We don't wanna benchmark JSON.stringify...
-      format: 'object',
+  if (!(process.platform === 'win32' && process.arch === 'ia32')) {
+    suite.add('combine 1000 maps using flatbuffers and stringify', async () => {
+      let map = new SourceMap();
+      for (let i = 0; i < 1000; i++) {
+        map.addBufferMappings(sourcemapBuffer, i * 4);
+      }
+      await map.stringify({
+        file: 'index.js.map',
+        sourceRoot: '/',
+        // We don't wanna benchmark JSON.stringify...
+        format: 'object',
+      });
+      map.delete();
     });
-    map.delete();
-  });
+  }
 
   suite.run();
 });

--- a/package.json
+++ b/package.json
@@ -15,6 +15,7 @@
     "compile-wasm": "make clean && make -j4",
     "transpile": "babel ./src/*.js --out-dir ./dist && flow-copy-source -v src dist",
     "prebuild": "prebuildify -t 10.20.1 --napi --strip --tag-libc",
+    "prebuild-ia32": "prebuildify -t 12.13.0 --napi --strip --tag-libc --arch=ia32",
     "build:dev": "node-gyp rebuild --debug",
     "rebuild": "rm -rf build && yarn build:dev",
     "rebuild-all": "yarn transpile && yarn compile-wasm && yarn rebuild",

--- a/package.json
+++ b/package.json
@@ -57,7 +57,7 @@
   },
   "dependencies": {
     "node-addon-api": "^3.0.0",
-    "node-gyp-build": "^4.2.2"
+    "node-gyp-build": "^4.2.3"
   },
   "devDependencies": {
     "@babel/cli": "^7.8.4",
@@ -72,7 +72,7 @@
     "husky": "4.2.5",
     "lint-staged": "^10.2.9",
     "mocha": "^7.1.2",
-    "prebuildify": "^4.0.0",
+    "prebuildify": "^4.1.1",
     "prettier": "^2.0.5",
     "source-map": "^0.7.3",
     "tiny-benchy": "^1.0.0"

--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "prebuild": "prebuildify -t 10.20.1 --napi --strip --tag-libc",
     "prebuild-ia32": "prebuildify -t 12.13.0 --napi --strip --tag-libc --arch=ia32",
     "build:dev": "node-gyp rebuild --debug",
-    "rebuild": "rm -rf build && yarn build:dev",
+    "rebuild": "shx rm -rf build && yarn build:dev",
     "rebuild-all": "yarn transpile && yarn compile-wasm && yarn rebuild",
     "install": "node-gyp-build",
     "prepublish": "npm run transpile",
@@ -25,7 +25,7 @@
     "typecheck": "flow",
     "format": "prettier --write \"./**/*.{js,md,mdx}\"",
     "compile-schema": "./flatc --cpp -o ./src ./src/sourcemap-schema.fbs",
-    "clean": "rm -rf dist build"
+    "clean": "shx rm -rf dist build"
   },
   "husky": {
     "hooks": {
@@ -74,6 +74,7 @@
     "mocha": "^7.1.2",
     "prebuildify": "^4.1.1",
     "prettier": "^2.0.5",
+    "shx": "^0.3.3",
     "source-map": "^0.7.3",
     "tiny-benchy": "^1.0.0"
   }

